v2/internal/testcommon/vcr/v3/tracking_roundtripper.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package v3
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon/vcr"
+)
+
+// Wraps an inner HTTP roundtripper to add a
+// counter for duplicated request URIs. This
+// is then used to match up requests in the recorder
+// - it is needed as we have multiple requests with
+// the same Request URL and it will return the first
+// one that matches.
+type requestCounter struct {
+	inner http.RoundTripper
+
+	countsMutex sync.Mutex
+	counts      map[string]uint32
+}
+
+func AddTrackingHeaders(inner http.RoundTripper) *requestCounter {
+	return &requestCounter{
+		inner:       inner,
+		counts:      make(map[string]uint32),
+		countsMutex: sync.Mutex{},
+	}
+}
+
+const (
+	COUNT_HEADER string = "TEST-REQUEST-ATTEMPT"
+	HASH_HEADER  string = "TEST-REQUEST-HASH"
+)
+
+var _ http.RoundTripper = &requestCounter{}
+
+func (rt *requestCounter) RoundTrip(req *http.Request) (*http.Response, error) {
+	if req.Method == "PUT" || req.Method == "POST" {
```

Comment:
Not every POST is guaranteed to have a body, it's optional. I _think_ that'll cause an issue because they'll all have the same hash?

PUTs I think probably are guaranteed to have a body.

Reply:
Tweaked the condition.